src/lib/static_variant_cast.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <type_traits>
+
+#include "boost/variant.hpp"
+
+#include "resolve_type.hpp"
+
+namespace opossum {
+
+template <typename Target>
+std::optional<Target> static_variant_cast(const AllTypeVariant& source) {
```

Comment:
```c++
/**
 * Contrary to the functions in lossless_cast.hpp, converts from an AllTypeVariant to any target, even if accuracy/data
 * is lost by the conversion.
 *
 * Use this function only in contexts where (query-)result accuracy is not affected (e.g., statistics/estimations)
 *
 * If @param source is NULL, return std::nullopt
 * If @param source is a string, perform a boost::lexical_cast<>
 * If @param source is arithmetic, perform a static_cast<>, clamping the returned value at
 *                                 `std::numeric_limits<Target>::min()/max()` to avoid undefined behaviour.
 */
template <typename Target>
std::optional<Target> static_variant_cast(const AllTypeVariant& source) {
```

and

```c++

/**
 * Same as `static_variant_cast()`, only that this variant Assert()s that @param source is not NULL
 */
template <typename Target>
Target non_null_static_variant_cast(const AllTypeVariant& source) {
```

Reply:
One more thought: Should we call this `lossy_variant_cast`? I worry that `static_...` does not generate enough attention when reading it somewhere else.

Reply:
Good idea. And I removed/inlined non_null_static_variant_cast because it was only used once

Reply:
Sounds good